src/Util/File.cc

```diff
@@ -79,6 +130,18 @@ fs::path SearchPath(std::string filename) {
     return fs::path();
 }
 
+/**
+ * @details This function attempts to identify the type of an image file by either checking
```

Comment:
Minor nitpick: I would suggest "either by checking... or by..." instead of "by either..." (and again in the next docstring).